src/number_theory/arithmetic_function.lean

```diff
@@ -248,6 +254,67 @@ instance [comm_ring R] : comm_ring (arithmetic_function R) :=
 { .. arithmetic_function.add_comm_group,
   .. arithmetic_function.comm_semiring }
 
+section module
+variables {M : Type*} [semiring R] [add_comm_monoid M] [semimodule R M]
+
+/-- The Dirichlet convolution of two arithmetic functions `f` and `g` is another arithmetic function
+  such that `(f * g) n` is the sum of `f x * g y` over all `(x,y)` such that `x * y = n`. -/
+instance : has_scalar (arithmetic_function R) (arithmetic_function M) :=
+⟨λ f g, ⟨λ n, ∑ x in divisors_antidiagonal n, f x.fst • g x.snd, by simp⟩⟩
```

Comment:
Can you weaken this at all, perhaps to
```suggestion
variables {M : Type*} [semiring R] [add_comm_monoid M]

/-- The Dirichlet convolution of two arithmetic functions `f` and `g` is another arithmetic function
  such that `(f * g) n` is the sum of `f x * g y` over all `(x,y)` such that `x * y = n`. -/
instance [mul_action R M] : has_scalar (arithmetic_function R) (arithmetic_function M) :=
⟨λ f g, ⟨λ n, ∑ x in divisors_antidiagonal n, f x.fst • g x.snd, by simp⟩⟩


variables {M : Type*} [semiring R] [add_comm_monoid M] [semimodule R M]
```
How strong a typeclass does `by simp` need here?